CMakeLists.txt

```diff
@@ -101,6 +101,11 @@ configure_file(icinga-spec-version.h.cmake icinga-spec-version.h)
 
 include(GetGitRevisionDescription)
 git_describe(GIT_VERSION --tags)
+include(git-archive-version.cmake)
+if(GIT_VERSION MATCHES "-NOTFOUND$" AND NOT GIT_ARCHIVE_VERSION MATCHES "Format")
+  set(GIT_VERSION "${GIT_ARCHIVE_VERSION}")
+endif()
```

Comment:
Do I understand correctly that this prefers `GetGitRevisionDescription` and `GIT_ARCHIVE_VERSION` set by `git archive` is only used as a fallback?

If both find a version, I'd prefer the one from `git archive` as if that's set, it means that the current source tree went through `git archive`. If `GetGitRevisionDescription` additionally finds commit information, odds are that this has been imported in some other git which probably doesn't have the version information we want here.

This would also mean that if `GIT_ARCHIVE_VERSION` was set, we wouldn't even have to bother `GetGitRevisionDescription`.